### ISSUE TITLE
Earthquake analysis

### ISSUE THREAD

**Issue by Charlotte-Morgan** [inasafe/inasafe#3654]
# problem
We need to ensure that legacy details are retained in the migration of InaSAFE3.5.3 earthquake analysis to the new version 4 architecture.

In version 3.5.x we had 3 earthquake impact functions for impact on population: pager model, ITB fatality model, ITB bayesian fatality model  and one for impact on buildings: earthquake building

![image](https://cloud.githubusercontent.com/assets/5743135/22139884/9d601d02-df1b-11e6-9df2-e4ae25fd1af6.png)

The earthquake_building analysis supported Nexis building data with specific attributes and building valuations .... 
![image](https://cloud.githubusercontent.com/assets/5743135/22139979/430a9250-df1c-11e6-8c8c-b4ad2af2a18f.png)

In version 4.0b2 we have a single impact function which offers 3 earthquake on population functions; however the names of these have changed from version 3.5.x. This could lead to confusion as the names are quite similar to each other

![image](https://cloud.githubusercontent.com/assets/5743135/22140095/e5a3932c-df1c-11e6-9c44-8be445bc4d8e.png)

![image](https://cloud.githubusercontent.com/assets/5743135/22140083/d5f8111e-df1c-11e6-81b9-05a62349449f.png)

![image](https://cloud.githubusercontent.com/assets/5743135/22140108/f8d7cfb2-df1c-11e6-9033-14aae3f18bdf.png)

# proposed solution 1
 - [x] update the bayesian_fatality_rates to be called ITB_bayesian_fatality_rates
![image](https://cloud.githubusercontent.com/assets/5743135/22140108/f8d7cfb2-df1c-11e6-9033-14aae3f18bdf.png)
- [ ] make sure ITB_bayesian fatality function is the one used by default in both InaSAFE desktop and InaSAFE realtime @lucernae 

# proposed solution 2
Ensure the legacy analysis on NEXIS building data with valuations is retained in the code

# cc
@timlinux @hadighasemi @dynaryu @RikkiWeber

**Comment by ismailsunni** [comment]
Hi @Charlotte-Morgan 

> Ensure the legacy analysis on NEXIS building data with valuations is retained in the code

Reading the code and recalling the history, this NEXIS is used to indicate whether the IF can run more calculation or not.

The work flow is:
1. The IF check whether there is attribute called `FLOOR_AREA`, `BUILDING_C`, and CONTENTS_C are in the exposure layer. It refers to a attribute for area of the building, the cost of the building, and the cost of the content of the building. Reference: https://github.com/inasafe/inasafe/blob/version-3_5_4/safe/impact_functions/earthquake/earthquake_building/impact_function.py#L104
2. If so, the IF will calculate for 2 more results:  building value and contents value. The IF will calculate for each building and sum the up. The formula is building_value = FLOOR_AREA * BUILDING_C and content_value = FLOOR_AREA * CONTENTS_C. Reference: https://github.com/inasafe/inasafe/blob/version-3_5_4/safe/impact_functions/earthquake/earthquake_building/impact_function.py#L142

Compared to InaSAE v4, this feature is the same as post processor with feature rate post processor (and similar one like cost post processor, value post processor, etc). It's also similar with the one that @fredychandra requested in the meeting on Wednesday.

Even better, in version 4, we do not need to have specific attribute name, but just need to map it in the wizard. The user also do not need to have size value, since it's computed automatically for building polygon. If you want to run it on building point, you need to specify which attribute is the size. We don't have one for this case in version 4. Is it common to have size attribute in building point layer? If so, I can add it to the wizard.

To sum up: Yes, this NEXIS can be supported in InaSAFE, in a better way.

**Comment by Charlotte-Morgan** [comment]
hi @ismailsunni  - oh this is wonderful. :)
Thankyou for explaining it so clearly.  @lrpower - you may be interested in this functionality in InaSAFE

**Comment by Gustry** [comment]
Proposed solution 1 is done in the PR above.

We just need some improvments we will be able to to do proposed solution 2 according to @ismailsunni.

**Comment by timlinux** [comment]
Just one note about this:

> The user also do not need to have size value, since it's computed automatically for building polygon. 

There may be a gotcha here if they have a single value for floor area pre-computed as ``levels * base floor area`` - we should provide similar logic if needed and also clarify different definitions between ``base_floor_area`` and ``total_floor_area``.

**Comment by Charlotte-Morgan** [comment]
thx for implementing solution 1

**Comment by timlinux** [comment]
Closing this as the realtime work has been moved to its own repo and a ticket has been opened there.
